apps/api/src/submission/entity/submission.entity.ts

```diff
@@ -19,4 +19,41 @@ export class SubmissionEntity extends BaseEntity {
   @Exclude()
   @Column('varchar', { nullable: false })
   version!: string;
+  @BeforeInsert()
+  beforeInsert() {
+    let { contactInformation, skillInformation, availabilityInformation } = this.payload;
+
+    // Remove specialties if the stream is non clinical
+    if (skillInformation.stream && skillInformation.stream !== streamsById.Nonclinical.id) {
```

Comment:
I think we'll need to also check for streams like midwife that shouldn't have specialties, something like a new helper in the common package `getStreamsWithoutSpecialties` could be use

Reply:
Actually this is captured by [EHPR-177](https://freshworks.atlassian.net/browse/EHPR-177), would be nice to add to reduce possible confusion in the future, what do you think?